Doc/library/typing.rst

```diff
@@ -1785,8 +1828,12 @@ These are not used in annotations. They are building blocks for declaring types.
 
       ``Point2D.__required_keys__`` and ``Point2D.__optional_keys__`` return
       :class:`frozenset` objects containing required and non-required keys, respectively.
-      Currently the only way to declare both required and non-required keys in the
-      same ``TypedDict`` is mixed inheritance, declaring a ``TypedDict`` with one value
+
+      Keys marked with ``Required`` will always appear in ``__required_keys__``
+      and keys marked with ``NotRequired`` will always appear in ``__optional_keys__``.
+
+      It is also possible to use mixed inheritance to declare both required and
```

Comment:
Maybe we should just drop this example, there's no reason to use it any more.

Reply:
I did add the more-specific prelude:

> For backwards compatibility with Python 3.10 and below, [...]